lighthouse-core/report/html/renderer/element-screenshot-renderer.js

```diff
@@ -0,0 +1,237 @@
+        background-image: url(${fullpageScreenshotUrl})
+      }`;
+    containerEl.appendChild(fullPageScreenshotStyleTag);
+  }
+
+  /**
+   * @param {DOM} dom
+   * @param {ParentNode} templateContext
+   * @param {LH.Artifacts.FullPageScreenshot} fullPageScreenshot
+   */
+  static installOverlayFeature(dom, templateContext, fullPageScreenshot) {
+    ElementScreenshotRenderer._installBackgroundImageStyle(dom, fullPageScreenshot);
+
+    for (const el of dom.document().querySelectorAll('.lh-element-screenshot')) {
+      el.addEventListener('click', () => {
+        const overlay = dom.createElement('div');
+        overlay.classList.add('lh-element-screenshot__overlay');
+        const clipRect = {
+          width: Number(el.getAttribute('rectWidth')),
+          height: Number(el.getAttribute('rectHeight')),
+          left: Number(el.getAttribute('rectLeft')),
+          right: Number(el.getAttribute('rectRight')),
+          top: Number(el.getAttribute('rectTop')),
+          bottom: Number(el.getAttribute('rectBottom')),
+        };
+        overlay.appendChild(ElementScreenshotRenderer.render(
+          dom,
+          templateContext,
+          clipRect,
+          fullPageScreenshot,
+          {
+            // TODO: should this be documentElement width?
+            width: window.innerWidth * 0.75,
+            height: window.innerHeight * 0.75,
+          }
+        ));
+        document.body.appendChild(overlay);
+        overlay.addEventListener('click', () => {
+          overlay.remove();
+        });
+      });
+    }
+  }
+
+  /**
+   * @param {DOM} dom
+   * @param {ParentNode} templateContext
+   * @param {LH.Artifacts.Rect} clipRect Region of screenshot to highlight.
+   * @param {LH.Artifacts.FullPageScreenshot} fullPageScreenshot
+   * @param {Size} viewportSize Size of region to render screenshot in.
+   * @return {Element}
+   */
+  static render(dom, templateContext, clipRect, fullPageScreenshot, viewportSize) {
+    const tmpl = dom.cloneTemplate('#tmpl-lh-element-screenshot', templateContext);
+    const containerEl = dom.find('.lh-element-screenshot', tmpl);
+
+    let zoomFactor = 1;
+    const viewport = {
+      width: viewportSize.width,
+      height: viewportSize.height,
+    };
+
+    containerEl.setAttribute('rectWidth', clipRect.width.toString());
+    containerEl.setAttribute('rectHeight', clipRect.height.toString());
+    containerEl.setAttribute('rectLeft', clipRect.left.toString());
+    containerEl.setAttribute('rectRight', clipRect.right.toString());
+    containerEl.setAttribute('rectTop', clipRect.top.toString());
+    containerEl.setAttribute('rectBottom', clipRect.bottom.toString());
+
+    // Zoom out (by double viewport but using 0.5 zoom factor) when highlighted region takes
+    // up most of the viewport. This provides more context for where on the page this element is.
+    /* todo: maybe only apply the width criterium in the preview screenshot (what's in the table, not the overlay) */
```

Comment:
is this relevant to keep?

Reply:
It was the original impl of zooming out for large images, kept for reference for you. L174-185 replaces it. I can remove it now.